lib/src/repository/traits/thermostat_repository.dart

```diff
@@ -25,6 +25,20 @@ class ThermostatRepository {
           new Set<AvailableThermostatMode>.from(
               trait.properties.availableThermostatModes);
 
+      final TemperatureRange? coolTempRange =
+          (trait.properties.coolSetPointRange != null)
```

Comment:
Brain is breaking... feels like this should use a null check instead of `!= null` but maybe not?